CollabApp/CollabApp.mvc/Controllers/PostController.cs

```diff
@@ -133,5 +137,74 @@ public IActionResult SortPosts(SortingOption sortBy)
             }
             return View("Posts", sortedPosts);
         }
+        [HttpGet]
+        public IActionResult Edit(int id)
+        {
+            var post = _context.Posts.FirstOrDefault(p => p.Id == id);
+            if (post == null)
+            {
+                return NotFound();
+            }
+
+            var currentUser = _httpContextAccessor.HttpContext.Session.GetString("Username");
+
+            if (currentUser != post.Author)
+            {
+                TempData["ErrorMessage"] = "You are not authorized to edit this post.";
+                return RedirectToAction("PostView", new { id });
+            }
+
+            return View(post);
+        }
+
+        [HttpPost]
+        public IActionResult Edit(int id, Post updatedPost)
+        {
+            var existingPost = _context.Posts.FirstOrDefault(p => p.Id == id);
+            if (existingPost == null)
+            {
+                return NotFound();
+            }
+
+            var currentUser = _httpContextAccessor.HttpContext.Session.GetString("Username");
+
+            if (currentUser != existingPost.Author)
+            {
+                TempData["ErrorMessage"] = "You are not authorized to edit this post.";
+                return RedirectToAction("PostView", new { id });
+            }
+
+            // Update the post properties with the changes
+            existingPost.Title = updatedPost.Title;
```

Comment:
If the edited title is empty, it errors. 
Use the extension method 'IsValidTitle()' to check for that.